### PR TITLE
Homework stats

### PR DESCRIPTION
**Pull request by Dantemss**
# Done:
- Added grade to GET plans API
- Decide who is responsible for hiding the grade for iReadings (have fun, FE)
- Fixed/improved specs

### CODE REVIEW

**Comment by Dantemss** [comment]
The code adds the grade for both HW and iReading. We should decide if the BE is responsible for hiding the iReading grade, or the FE.

**Comment by Dantemss** [comment]
Also need to decide on the value returned if there are no gradable (completed) exercises. Could be nil (i.e. key is absent from JSON, 0 or 100).

File: app/representers/api/v1/task_plan_representer.rb [hunk 1]
```diff
@@ -39,8 +39,9 @@ class TaskPlanRepresenter < Roar::Decorator
              writeable: true
 
     property :stats,
-             extend: IReadingStatsRepresenter,
-             getter: ->(args){ args[:stats] },
+             extend: TaskPlanStatsRepresenter,
+             getter: ->(args){ CalculateTaskPlanStats.call(plan: self).outputs.stats },
```
Comment (jpslav) [inline app/representers/api/v1/task_plan_representer.rb]:
```ruby
getter: -> (*) { CalculateTaskPlanStats[plan: self] }
```

**Comment by jpslav** [comment]
I'm ok with the BE providing extra info that the FE doesn't show.

File: app/representers/api/v1/task_plan_representer.rb (hunk 1 above)
Comment (Dantemss) [inline app/representers/api/v1/task_plan_representer.rb]: :)

**Comment by Dantemss** [comment]
@jpslav ready for review

**Comment by jpslav** [comment]
Looks good @Dantemss, gonna merge.  In the back of your mind can you start thinking about how to cache this stuff so we don't have to compute from scratch every time?

**Pull request merged**